askama_derive/src/generator.rs

```diff
@@ -1210,66 +1210,106 @@ impl<'a> Generator<'a> {
         Ok(DisplayWrap::Wrapped)
     }
 
+    #[cfg(not(feature = "serde-json"))]
+    fn _visit_json_filter(
+        &mut self,
+        _: &mut Buffer,
+        _: &[Expr<'_>],
+    ) -> Result<DisplayWrap, CompileError> {
+        Err("the `json` filter requires the `serde-json` feature to be enabled".into())
+    }
+
+    #[cfg(feature = "serde-json")]
+    fn _visit_json_filter(
+        &mut self,
+        buf: &mut Buffer,
+        args: &[Expr<'_>],
+    ) -> Result<DisplayWrap, CompileError> {
+        if args.len() != 1 {
+            return Err("unexpected argument(s) in `json` filter".into());
+        }
+        buf.write("::askama::filters::json(");
+        self._visit_args(buf, args)?;
+        buf.write(")?");
+        Ok(DisplayWrap::Unwrapped)
+    }
+
+    #[cfg(not(feature = "serde-yaml"))]
+    fn _visit_yaml_filter(
+        &mut self,
+        _: &mut Buffer,
+        _: &[Expr<'_>],
+    ) -> Result<DisplayWrap, CompileError> {
+        Err("the `yaml` filter requires the `serde-yaml` feature to be enabled".into())
+    }
+
+    #[cfg(feature = "serde-yaml")]
+    fn _visit_yaml_filter(
+        &mut self,
+        buf: &mut Buffer,
+        args: &[Expr<'_>],
+    ) -> Result<DisplayWrap, CompileError> {
+        if args.len() != 1 {
+            return Err("unexpected argument(s) in `safe` filter".into());
+        }
+        buf.write("::askama::filters::yaml(");
+        buf.write(self.input.escaper);
+        buf.write(", ");
+        self._visit_args(buf, args)?;
+        buf.write(")?");
+        Ok(DisplayWrap::Wrapped)
+    }
```

Comment:
Wouldn't it be simpler to instead have:

```suggestion
    fn _visit_yaml_filter(
        &mut self,
        buf: &mut Buffer,
        args: &[Expr<'_>],
    ) -> Result<DisplayWrap, CompileError> {
        if cfg!(not(feature = "serde-yaml")) {
            return Err("the `yaml` filter requires the `serde-yaml` feature to be enabled".into());
        }
        if args.len() != 1 {
            return Err("unexpected argument(s) in `safe` filter".into());
        }
        buf.write("::askama::filters::yaml(");
        buf.write(self.input.escaper);
        buf.write(", ");
        self._visit_args(buf, args)?;
        buf.write(")?");
        Ok(DisplayWrap::Wrapped)
    }
```

?

Reply:
Yes, that's much better!